### ISSUE TITLE
[bug] conan 2.0.14 regression on core.sources:download_urls

### ISSUE THREAD

**Issue by jonboan** [conan-io/conan#15103]
### Environment details

* Operating System+version: ubuntu-22.04
* Compiler+version: gcc11
* Conan version: 2.0.14
* Python version: 3.11.2

### Steps to reproduce

There  is a regression from 2.0.13 -> 2.0.14. 
We make use of the 
`core.sources:download_urls`  in our global.conf configuration and noticed with an update to 2.0.14 that our packages from CCI will no longer build as expected. 
 Steps to reproduce 
1. global.conf that makes use of `core.sources:download_urls
```
core.sources:download_urls=["http://XXXXX/artifactory/conan-center-sources/", "origin"]
```
2.  conan 2.0.13 - `conan install --require zlib/1.3 --build=zlib/1.3` will build with no problems. 
3.  conan 2.0.14 - `conan install --require zlib/1.3 --build=zlib/1.3` will fail under 2.0.14 with error
```
ERROR: zlib/1.3: Error in source() method, line 51
	get(self, **self.conan_data["sources"][self.version],
	AttributeError: 'SourcesCachingDownloader' object has no attribute 'conan_api'

```
There is no change in the global.conf. A representative ones used that causes the problem is 
Logs are attached for the 2.0.13 and 2.0.14 

### Logs

__CONAN 2.0.13 __
```
conan install --require zlib/1.3 --build=zlib/1.3 

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
[conf]
tools.build:jobs=12

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
[conf]
tools.build:jobs=12


======== Computing dependency graph ========
Graph root
    cli
Requirements
    zlib/1.3#06023034579559bb64357db3a53f88a4 - Cache

======== Computing necessary packages ========
zlib/1.3: Forced build from source
Requirements
    zlib/1.3#06023034579559bb64357db3a53f88a4:b647c43bfefae3f830561ca202b6cfd935b56205 - Build

======== Installing packages ========
zlib/1.3: WARN: Trying to remove corrupted source folder
zlib/1.3: WARN: This can take a while for big packages
zlib/1.3: Calling source() in /home/boanj/.conan2/p/zlib345774da50bf7/s/src
zlib/1.3: Sources for ['https://zlib.net/fossils/zlib-1.3.tar.gz', 'https://github.com/madler/zlib/releases/download/v1.3/zlib-1.3.tar.gz'] found in remote backup http://XXXXX/artifactory/conan-center-sources/

-------- Installing package zlib/1.3 (1 of 1) --------
zlib/1.3: Building from source
zlib/1.3: Package zlib/1.3:b647c43bfefae3f830561ca202b6cfd935b56205
zlib/1.3: Copying sources to build folder
zlib/1.3: Building your package in /home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/b
zlib/1.3: Calling generate()
zlib/1.3: Generators folder: /home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/b/build/Release/generators
zlib/1.3: CMakeToolchain generated: conan_toolchain.cmake
zlib/1.3: CMakeToolchain generated: CMakePresets.json
zlib/1.3: CMakeToolchain generated: ../../../src/CMakeUserPresets.json
zlib/1.3: Generating aggregated env files
zlib/1.3: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
zlib/1.3: Calling build()
zlib/1.3: Apply patch (conan): separate static/shared builds, disable debug suffix, disable building examples
zlib/1.3: Running CMake.configure()
zlib/1.3: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/b/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/p" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/b/src"
-- Using Conan toolchain: /home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The C compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of off64_t
-- Check size of off64_t - done
-- Looking for fseeko
-- Looking for fseeko - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Renaming
--     /home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/b/src/zconf.h
-- to 'zconf.h.included' because this file is included with zlib
-- but CMake generates it automatically in the build directory.
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0091


-- Build files have been written to: /home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/b/build/Release

zlib/1.3: Running CMake.build()
zlib/1.3: RUN: cmake --build "/home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/b/build/Release" -- -j12
[  6%] Building C object CMakeFiles/zlib.dir/compress.c.o
[ 12%] Building C object CMakeFiles/zlib.dir/crc32.c.o
[ 25%] Building C object CMakeFiles/zlib.dir/adler32.c.o
[ 25%] Building C object CMakeFiles/zlib.dir/gzclose.c.o
[ 31%] Building C object CMakeFiles/zlib.dir/deflate.c.o
[ 37%] Building C object CMakeFiles/zlib.dir/gzlib.c.o
[ 43%] Building C object CMakeFiles/zlib.dir/gzread.c.o
[ 50%] Building C object CMakeFiles/zlib.dir/gzwrite.c.o
[ 56%] Building C object CMakeFiles/zlib.dir/inflate.c.o
[ 62%] Building C object CMakeFiles/zlib.dir/inftrees.c.o
[ 68%] Building C object CMakeFiles/zlib.dir/infback.c.o
[ 75%] Building C object CMakeFiles/zlib.dir/inffast.c.o
[ 81%] Building C object CMakeFiles/zlib.dir/trees.c.o
[ 87%] Building C object CMakeFiles/zlib.dir/uncompr.c.o
[ 93%] Building C object CMakeFiles/zlib.dir/zutil.c.o
[100%] Linking C static library libz.a
[100%] Built target zlib

zlib/1.3: Package 'b647c43bfefae3f830561ca202b6cfd935b56205' built
zlib/1.3: Build folder /home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/b/build/Release
zlib/1.3: Generating the package
zlib/1.3: Packaging in folder /home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/p
zlib/1.3: Calling package()
zlib/1.3: Running CMake.install()
zlib/1.3: RUN: cmake --install "/home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/b/build/Release" --prefix "/home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/p"
-- Install configuration: "Release"
-- Installing: /home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/p/lib/libz.a
-- Installing: /home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/p/include/zconf.h
-- Installing: /home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/p/include/zlib.h

zlib/1.3: package(): Packaged 1 '.a' file: libz.a
zlib/1.3: package(): Packaged 1 file: LICENSE
zlib/1.3: package(): Packaged 2 '.h' files: zconf.h, zlib.h
zlib/1.3: Created package revision 0d9d3d9636d22ee9f92636bb1f92958b
zlib/1.3: Package 'b647c43bfefae3f830561ca202b6cfd935b56205' created
zlib/1.3: Full package reference: zlib/1.3#06023034579559bb64357db3a53f88a4:b647c43bfefae3f830561ca202b6cfd935b56205#0d9d3d9636d22ee9f92636bb1f92958b
zlib/1.3: Package folder /home/boanj/.conan2/p/b/zlibda2aa3a8d59e5/p
WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN: deprecated:     'cpp_info.names' used in: zlib/1.3

======== Finalizing install (deploy, generators) ========
cli: Generating aggregated env files
cli: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
Install finished successfully

```

__CONAN 2.0.14__
```
conan install --require zlib/1.3 --build=zlib/1.3 

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
[conf]
tools.build:jobs=12

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
[conf]
tools.build:jobs=12


======== Computing dependency graph ========
Graph root
    cli
Requirements
    zlib/1.3#06023034579559bb64357db3a53f88a4 - Cache

======== Computing necessary packages ========
zlib/1.3: Forced build from source
Requirements
    zlib/1.3#06023034579559bb64357db3a53f88a4:b647c43bfefae3f830561ca202b6cfd935b56205 - Build

======== Installing packages ========
zlib/1.3: Sources downloaded from 'conan-center-index'
zlib/1.3: Calling source() in /home/boanj/.conan2/p/zlib345774da50bf7/s/src
ERROR: zlib/1.3: Error in source() method, line 51
	get(self, **self.conan_data["sources"][self.version],
	AttributeError: 'SourcesCachingDownloader' object has no attribute 'conan_api'

```
```

**Comment by AbrilRBS** [comment]
Hi @jonboan thanks a lot for your report. We've identified the issue and will fix it asap. For now to unblock you until we release the fix, this only reproduces when the `core.sources:download_cache` conf is not set, so you might try setting it to something that works for you (It's usually set to `<CONAN_HOME>/sources` if the conf is not set)

Will ping you when this is fixed

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/15109 merged, it will be released in next 2.0.15, thanks very much for reporting.
